src/macaron/dependency_analyzer/cyclonedx.py

```diff
@@ -0,0 +1,206 @@
+# Copyright (c) 2023 - 2023, Oracle and/or its affiliates. All rights reserved.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl/.
+
+"""This module contains helper functions to process CycloneDX SBOM."""
+
+import json
+import logging
+import os
+from collections.abc import Iterable
+from pathlib import Path
+from typing import Optional
+
+from macaron.config.global_config import global_config
+from macaron.dependency_analyzer.dependency_resolver import DependencyAnalyzer, DependencyInfo
+from macaron.errors import MacaronError
+from macaron.output_reporter.scm import SCMStatus
+
+logger: logging.Logger = logging.getLogger(__name__)
+
+
+class CycloneDXParserError(MacaronError):
+    """The CycloneDX error class."""
+
+
+def deserialize_bom_json(file_path: Path) -> dict:
+    """Deserialize the bom.json file.
+
+    Parameters
+    ----------
+    file_path : str
+        Path to the bom.json file.
+
+    Returns
+    -------
+    dict
+        The BOM content as a dictionary.
+
+    Raises
+    ------
+    CycloneDXParserError
+        If the bom.json file cannot be located or deserialized.
+
+    """
+    # TODO: use the official CycloneDX library to validate and deserialize BOM files
+    # once merged: https://github.com/CycloneDX/cyclonedx-python-lib/pull/290
+    if not os.path.exists(file_path):
+        raise CycloneDXParserError(f"Unable to locate the BOM file: {str(file_path)}.")
+
+    with open(file_path, encoding="utf8") as file:
+        try:
+            return dict(json.load(file))
+        except (ValueError, json.JSONDecodeError) as error:
+            raise CycloneDXParserError(f"Could not process the dependencies at {file_path}: {error}") from None
```

Comment:
I wonder if it is better to raise the error as follow instead of appending the error message.
```suggestion
            raise CycloneDXParserError(f"Could not process the dependencies at {file_path}.") from error
```
This syntax is according to [PEP-3134](https://legacy.python.org/dev/peps/pep-3134/).

Reply:
The reason I'm using `None` is to avoid adding the previous error stack traces as the cause. That gets really messy. See the docs [here](https://docs.python.org/3/library/exceptions.html#exception-context).